### PR TITLE
Add db migration for CorePasswordCredential

### LINKED ISSUE #4007: task: Add Migration to move credential from GitRepository into CorePasswordCredential
### Component

API Server / GraphQL

### Task Description

#3945 updated the format of the GitRepository model and introduced a new model `CorePasswordCredential` to store credential
We need to write a migration to convert existing GitRepository node into the new format

### PR DESCRIPTION
**Pull request by dgarros**
Fixes #4007

The migration will automatically create a new `CorePasswordCredential` node for each `GitGenericRepository` and it will move the attributes : `username` and `password` to the new node.

The new `CorePasswordCredential` will have the same name as the `GitGenericRepository`

As part of this migration, I've also added a step to initialize `admin_status` to `active` on existing repository

### CODE REVIEW

File: backend/infrahub/core/migrations/query/attribute_add.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    async def query_init(self, db: InfrahubDatabase, **kwargs: dict[str, Any]) -> None:
+        branch_filter, branch_params = self.branch.get_query_filter_path(at=self.at.to_string())
+        self.params.update(branch_params)
+
+        self.params["node_kind"] = self.node_kind
+        self.params["attr_name"] = self.attribute_name
+        self.params["attr_type"] = self.attribute_kind
+        self.params["branch_support"] = self.branch_support
+
+        if self.default_value:
```
Comment (ogenstad) [inline backend/infrahub/core/migrations/query/attribute_add.py:41]: `if self.default_value is not None`?

**Review by ogenstad — APPROVED** [review]
LGTM.

A minor comment while I don't think it matters for this migration and where we are now, with this we'd remove the history of old credentials. Not really relevant for this type of data but it could be something to consider for future migrations where the data is different.

**Comment by dgarros** [comment]
> A minor comment while I don't think it matters for this migration and where we are now, with this we'd remove the history of old credentials. Not really relevant for this type of data but it could be something to consider for future migrations where the data is different.

it was a good surprise to me as well, the migration is taking care of the history because we are moving the Attribute node with all its values :)

File: backend/infrahub/core/migrations/query/attribute_add.py (hunk 1 above)
Comment (ogenstad) [reply backend/infrahub/core/migrations/query/attribute_add.py:41]: I might have been a bit unclear in the first comment but I think this should be changed. If the default value is `0` we will set it to `NULL_VALUE` instead with this.

**Pull request merged**